### PR TITLE
Bypass document validation option (v >= 3.2) and index support

### PR DESCRIPTION
**Pull request by skobets**
https://docs.mongodb.org/manual/core/document-validation/

### CODE REVIEW

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/UpdateOptions.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 @DataObject
 public class UpdateOptions {
 
-  /**
+    /**
```
Comment (karianna) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/UpdateOptions.java]: really minor, but why the indent here?

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java [hunk 2]
```diff
@@ -52,8 +54,9 @@
 
   private static final UpdateOptions DEFAULT_UPDATE_OPTIONS = new UpdateOptions();
   private static final FindOptions DEFAULT_FIND_OPTIONS = new FindOptions();
-  private static final String ID_FIELD = "_id";
+  public static final IndexOptions DEFAULT_INDEX_OPTIONS = new IndexOptions();
```
Comment (karianna) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java]: Could be private?

**Comment by karianna** [comment]
Thanks for the PR - looking good so far!  We'd ideally like to see some accompanying tests with the 3 new index methods you added.  We also need the contributor agreement to be signed.

**Comment by skobets** [comment]
Hello, I removed remarks, and added tests.

**Comment by karianna** [comment]
@skobets Thanks for the updated PR - I'm absolutely rammed for time so will review on Monday (apologies for the long process here, we really do appreciate the contribution!).  One other thing to check, have you signed the Vert.x contributor agreement?

**Comment by skobets** [comment]
@karianna Yes, I signed an agreement, according to the instructions:
[https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md](https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md) 
I have not finished yet PR and continue to improve it.

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/IndexOptions.java [hunk 3]
```diff
@@ -12,225 +13,212 @@
...
+
+  public IndexOptions() {
+    wrap = new com.mongodb.client.model.IndexOptions();
+  }
+
+  public IndexOptions(IndexOptions other) {
+    wrap = other.unwrap();
+  }
+
+  // TODO not implemented
```
Comment (karianna) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/IndexOptions.java]: Remove TODO?

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/IndexOptions.java [hunk 4]
```diff
@@ -12,225 +13,212 @@
...
+  public IndexOptions(JsonObject json) {
+    wrap = new com.mongodb.client.model.IndexOptions();
+  }
+
+  // TODO not implemented
+  public JsonObject toJson() {
```
Comment (karianna) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/IndexOptions.java]: Remove TODO?

File: vertx-mongo-client/src/main/java/io/vertx/ext/mongo/Indexes.java [hunk 5]
```diff
@@ -2,30 +2,68 @@
...
+
+  public static JsonObject compoundIndex(JsonObject... indexes) {
+    return compoundIndex(Arrays.asList(indexes));
+  }
+
+
+  public static JsonObject compoundIndex(List<JsonObject> indexes) {
+    JsonObject jsonObject = new JsonObject();
+    for (JsonObject index : indexes) {
+      Set<String> strings = index.fieldNames();
```
Comment (karianna) [inline vertx-mongo-client/src/main/java/io/vertx/ext/mongo/Indexes.java]: Improve naming? What sort of Strings?

**Comment by karianna** [comment]
Couple of minor comments, but apart from that it looks good!

**Comment by karianna** [comment]
Hi @skobets - are you able to update this PR to the latest code and review the last few comments I made?

**Comment by karianna** [comment]
Hi @skobets - This PR will need to be rebased and the index support removed (we merged in a separate PR for this).  Are you able to update this PR?

**Comment by karianna** [comment]
@skobets I'll leave this open for another week and then close as out of date.

**Pull request closed without merging**